test/Network/Packet.test.cpp

```diff
@@ -22,15 +22,15 @@
         CHECK(packet.getData() != nullptr);                  \
         CHECK(packet.getDataSize() == sizeof(expected));     \
         CHECK(!packet.endOfPacket());                        \
-        CHECK(static_cast<bool>(packet));                    \
+        CHECK(bool{packet});                                 \
```

Comment:
Didn't we decide against this style of casting?

Reply:
Not that I recall. In this case `bool{packet}` is performing a lossless cast because this syntax is forbidden from doing anything that throws away information. Because `sf::Packet` defines `operator bool` you can convert an `sf::Packet` to a boolean without a static cast. This `{}` syntax is less powerful than a static cast and we like to use the least powerful tool that solves the problem.

Reply:
Hmm, I'm not too sure about this PR. 

`static_cast` denotes an ambiguous conscious decision to trigger a conversion. It adds clarity of intention. A side benefit is that it's also easier to spot/search every conversion, as you can simply grep for `static_cast`.

We would also go against this core guideline: [ES.49: If you must use a cast, use a named cast](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts-named).

This other guideline ([https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es64-use-the-tenotation-for-construction](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es64-use-the-tenotation-for-construction)) also suggests that `T{...}` should be used for construction, not conversion. 

I know that construction can be conversion, but semantically speaking constructing something and converting something are two different intentions that can be conveyed with separate tools, `T{...}` and `static_cast<T>(...)` respectively.

If we were to go down this route, why *ever* use `static_cast<T>(...)` instead of `T{...}`? What would be our policy to prefer one over the other?

I'd rather see an issue first discussing the pros/cons of both approaches, and then we can create a policy that we'd apply library-wide instead of in a few selected cases.